### PR TITLE
Add visa sponsorship message when course can sponsor visa

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

Now that we sync with Publish we can use the attributes from the course in the candidate interface.

We should now not display the visa sponsorship message if the course can sponsor visa for students and skilled workers.

### CODE REVIEW

File: app/components/candidate_interface/course_choices_review_component.rb [hunk 1]
```diff
@@ -268,6 +269,15 @@ def application_predates_visa_sponsorship_information?(application_choice)
...
     end
 
+    def course_can_sponsor_visa?(application_choice)
+      (
+        application_choice.course.salary? && application_choice.course.can_sponsor_skilled_worker_visa?
+      ) ||
+        (
+          !application_choice.course.salary? && application_choice.course.can_sponsor_student_visa?
+        )
+    end
```
Comment (tomas-stefano) [inline app/components/candidate_interface/course_choices_review_component.rb:279]: I use a similar logic from the provider_can_sponsor_visa? below. Let me know if you prefer to have only one method that vqries between course and provider (e.g `can_sponsor_visa?(application_choice, :course) ||  can_sponsor_visa?(application_choice, :provider)`. I am happy to change if that's the preferred approach.

**Comment by tomas-stefano** [comment]
[Add Visa sponsorship on Courses coming from Find/Pub](https://trello.com/c/HdxpKqNH/1300-add-visa-sponsorship-on-courses-coming-from-find-pub)

File: app/components/candidate_interface/course_choices_review_component.rb [hunk 2]
```diff
@@ -251,6 +251,7 @@ def interview_row(application_choice)
     def visa_details_row(application_choice)
       return if (immigration_right_to_work?(application_choice) ||
                 application_predates_visa_sponsorship_information?(application_choice)) ||
+                course_can_sponsor_visa?(application_choice) ||
                 provider_can_sponsor_visa?(application_choice)
```
Comment (frankieroberto) [inline app/components/candidate_interface/course_choices_review_component.rb:255]: I think we can drop the `provider_can_sponsor_visa?` check now, and rely entirely on the course setting?

```suggestion
                course_can_sponsor_visa?(application_choice)
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]

File: app/components/candidate_interface/course_choices_review_component.rb (hunk 2 above)
Comment (tomas-stefano) [reply app/components/candidate_interface/course_choices_review_component.rb:255]: Ok, I was in doubt if I should use as failsafe mechanism but happy to remove.

Comment (tomas-stefano) [reply app/components/candidate_interface/course_choices_review_component.rb]: Done

**Review by frankieroberto — APPROVED** [review]

**Review by stevehook — APPROVED** [review]
Makes sense to me.

One question: is there another ticket to add the two new attributes to the course show page in support? (`support/courses/:id`)

**Comment by tomas-stefano** [comment]
@stevehook good catch. I created and moved to triage https://trello.com/c/Yt55k8yM/1373-add-visa-sponsorship-for-courses-in-support-interface

**Comment by frankieroberto** [comment]
@tomas-stefano we could also probably remove the provider-level attributes (and stop syncing them) once this is merged, as we would no longer need those (and they're bound to just confuse future-us)...?

**Pull request merged**